### ISSUE TITLE
RSU.java doesnt work

### ISSUE THREAD

**Issue by parthjoshi09** [VanetSim/VanetSim#1]
I am new to java and VanetSIm so dont know whether this is an issue or not but I tried editing the existing RSU.java file with the one found here .>> https://github.com/VanetSim/VanetSim/blob/0e6f24d75c2c02704c736b2b9586f24a34e71c36/src/vanetsim/scenario/RSU.java

Now I dont know if we are supposed to replace or both the files are same. But I tried to complie RSU.java gven on github I got like 100 errors. So is the code wrong or am i doing something wrong here?

**Comment by VanetSim** [comment]
Hi! Without a proper error description I might not be able to help you. The code itself works fine. Did you check out the complete branch and re-compiled the project or did you try to mix files from different branches? Could you please name some of the 100 errors?

**Comment by parthjoshi09** [comment]
Okay so let me tell you what I did exactly.
I opened the RSU.java file in gedit mode and next I copied everything from
the raw RSU.java file from the github.
Then I tried to compile only the RSU.java file inside the scenario folder
and it said to have 100 errors.

Now I dont exactly know what you mean when you say ''check out the complete
branch and re-compiled the project or did you try to mix files from
different branches? "

Also as per your request I am attaching screenshots of last ew errors.
Please tell me what I am doing wrong or how to run these files

**Comment by parthjoshi09** [comment]
Can you help me please? Its kinda urgent.

**Comment by VanetSim** [comment]
I am not sure where I can find your screenshot. But generally speaking: If I understand correctly you used a pre existing package of the VANETSim and just exchanged the content of the RSU.java. This will not work because the different classes are not independent from each other. Other classes might be changed in a later version of the VANETSim and the new RSU.class might access these classes. If you just exchange one class some information might be missing. In general, the VANETSim is probably not a good point to start learning java or programming in general. Which VANETSim package did you use as the starting point?

**Comment by parthjoshi09** [comment]
I used VanetSim version 1.3 as found here.
http://svs.informatik.uni-hamburg.de/vanet/#

Also I am not trying to learn java. I downloaded VanetSim just so to
simulate vanet scenarios. I simulated the inbuilt examples of NewYork
Berlin successfully. But those those scenarios doesnt contain
Vehicle-to-Infrastructure mode. So when I saw a file named RSU I just got
excited and tried to run it. So I would glad if we can make it work, but if
it doesnt run, thats okay.
Also I dont know if you developed vanetsim or not, but is there anything
else I can do with the inbuilt examples or I will have to write my own
codes?

**Comment by VanetSim** [comment]
Yes I was one of the developers of VANETSim. As I am no longer a student I had to quit the work on this project of mine. Therefore, I cannot support you with coding because of a lack of time. The updated version of the RSU class provides ways to triangulate cars. I am guessing that VANETSim does not provide the functionality you need. But to be on the safe side: What do you want to achieve with the Vehicle to Infrastructure functionality? There are some functionalities build in and it could be enough to create your own logger (depending on what you want to do). For example rule based detection of attackers, concepts to protect the users privacy and some other features are already implemented. VANETSim is specialized in IT security.

**Comment by parthjoshi09** [comment]
First of all thanks a lot for replying.
I am trying to achieve vehicle-to-vehicle as well as
vehicle-to-infrastucture simultaneously. I have been trying that in ns3 but
had unsatisfactory results. So I tried VanetSim. Those scenarios of Newyork
and Berlin, I can make them work as V2V mode of communication. But I also
want my vehicles to communicate with RSUs at the same time too. Hence the
issue.
Also you mentioned some inbuilt functionalities. So how exactly do I
simulate all those java files inside the scenario folder? And would I be
able to integrate them with existing .xml files?

**Comment by VanetSim** [comment]
If you want to analyze something regarding the network layer I highly recommend using ns2 or ns3 with some VANET addon. If you are more interested in the movement and the application layer or security issues VANETSim might be applicable for you. VANETSim supports V2V and V2I but is limited on the network protocols. For communication with RSUs (V2I) the messages of the vehicles are forwarded into the range of the RSU. Beacons are not forwarded (would not make sense I think). This functionality is coded in the sendMessages() function in the Vehicle.java class. But this is already implemented in the version 1.3.. The xml files are only map files or scenario files. You can import any open street map and VANETSim can create a map out of it (an example should be available on the website). The scenario files are the settings of the simulation. You can open a map and for example create different cares or place some mix zones. These settings can be saved into a xml scenario file.

**Comment by parthjoshi09** [comment]
I have done exaclty like you said. I created a map of my own city through
OpenStreetMap, loaded it into VanetSim, created a different scenario where
I defined the no. of vehicles, there velocities, mac range and other
variables. That part is ok.
What I want to know is how to simulate, if we can, those java files inside
the scenarios folder like Vehicle.java, Attackers.java and all that.

**Comment by VanetSim** [comment]
This files are already in use. For example Vehicle.java includes the logic behind the vehicles (e.g., how they move). The RSU.java contains the logic of the RSU, which can also be placed like vehicles. The reason why you don't see the *.class files is because they are all inside the VanetSimStarter.jar. It was created with an eclipse plugin called "fat jar". Depending on your operating system you can extract the jar and look inside. There you will find the *.class files.

**Comment by parthjoshi09** [comment]
Oh so those files are actually in use. I thought that we can simulate
something different using them.
So it seems my whole issue with VanetSim was pointless. What a lol moment.
Looks like back to ns3 for v2i communication.

Anyway thanks a lot man. You dont expect many developers around here to
actually have patience and explain/solve the difficulties of    users.
Never caught your name but..
I am Parth, pursuing masters in communications engineering in India.
Again thanks.

**Comment by VanetSim** [comment]
You are welcome. My name is Andreas. I wish you the best for your studies.

**Comment by geovidal** [comment]
Hello geeks, I saw the maps included in the downloads, NewYork, Berlin etc. I'm trying to get an openstreetmap XML file for my city too. I need help on how to do that. Thanks in anticipation
